src/libvfcinstrument/libVFCInstrument.cpp

```diff
@@ -43,11 +43,38 @@
   (Builder.CreateCall3(func, op1, op2, op3, ""))
 #define CREATE_CALL2(func, op1, op2) (Builder.CreateCall2(func, op1, op2, ""))
 #define CREATE_STRUCT_GEP(t, i, p) (Builder.CreateStructGEP(i, p))
+/* This function must be used with at least one variadic argument otherwise */
+/* it will fails when compiling since it will expand as
+ * M.getOrInsertFunction(name,res,,(Type*)NULL) */
+/* It could be fixed when __VA_OPT__ will be available (see
+ * https://gcc.gnu.org/onlinedocs/cpp/Variadic-Macros.html)*/
+#define GET_OR_INSERT_FUNCTION(M, name, res, ...)                              \
+  M.getOrInsertFunction(name, res, __VA_ARGS__, (Type *)NULL)
+typedef llvm::Constant *_LLVMFunctionType;
+#elif LLVM_VERSION_MAJOR < 5
+#define CREATE_CALL3(func, op1, op2, op3)                                      \
+  (Builder.CreateCall(func, {op1, op2, op3}, ""))
+#define CREATE_CALL2(func, op1, op2) (Builder.CreateCall(func, {op1, op2}, ""))
```

Comment:
CREATE_CALL2 seems always the same in these #ifdef blocks ?
Should we keep it here ?

Reply:
Yes, it is. I prefer that each version has its definition for readability purposes, even it is the same. But we can factorize it if you prefer.

Reply:
Ok we can keep it like that.